include/onnxruntime/core/session/onnxruntime_ep_c_api.h

```diff
@@ -169,6 +225,75 @@ struct OrtEpApi {
    * \since Version 1.23.
    */
   ORT_API_T(const char*, NodeComputeContext_NodeName, _In_ const OrtNodeComputeContext* context);
+
+  /** \brief Register an allocator with the OrtEpDevice.
+   *
+   * This allows an EP to provide OrtMemoryInfo for DEFAULT and HOST_ACCESSIBLE memory types as needed.
```

Comment:
nit: could we clarify that this only allows the EP to provide one DEFAULT memory type and one HOST_ACCESSIBLE memory type. The implementation would keep replacing memory infos if this function is called multiple times with HOST_ACCESSIBLE (for example).